pkg/kubelet/kubelet.go

```diff
@@ -2871,6 +2872,30 @@ func (kl *Kubelet) LatestLoopEntryTime() time.Time {
 	return val.(time.Time)
 }
 
+// emitRuntimeConditionEvent emits an event about optional runtime conditions for informative purpose.
+func (kl *Kubelet) emitOptionalRuntimeConditionEvent(conditions []kubecontainer.RuntimeCondition) {
+	var ss []string
+	for _, cond := range conditions {
+		switch cond.Type {
+		case kubecontainer.RuntimeReady, kubecontainer.NetworkReady:
+			// NOP, as they are not optional.
+		default:
+			// https://github.com/kubernetes/cri-api/blob/v0.29.1/pkg/apis/runtime/v1/api.proto#L1496-L1517
+			// > 2. Optional conditions: Conditions are informative to the user, but kubelet
+			// > will not rely on. Since condition type is an arbitrary string, all conditions
+			// > not required are optional. These conditions will be exposed to users to help
+			// > them understand the status of the system.
+			if !cond.Status {
+				ss = append(ss, fmt.Sprintf("%s is unsatisfied: %s: %q", string(cond.Type), cond.Reason, cond.Message))
```

Comment:
Would it make sense to make sure condition Reason/Message are unique or it's guaranteed?